modules/custom/advanced_help_block/src/Form/AdvancedHelpBlockSettingsForm.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Drupal\advanced_help_block\Form;
+
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+
+/**
+ * Class AdvancedHelpBlockSettingsForm.
+ *
+ * @package Drupal\advanced_help_block\Form
+ * @ingroup advanced_help_block
+ */
+class AdvancedHelpBlockSettingsForm extends FormBase {
```

Comment:
No need to create a dedicated form, use block form from core, please http://cgit.drupalcode.org/examples/tree/block_example/src/Plugin/Block/ExampleConfigurableTextBlock.php

Reply:
This is not configuration for block, as you recommended. This is configuration form for module. I think this recommendation is misunderstanding. Not fixed, Please check again.